packages/cadl-vscode/test/namespace.ts

```diff
@@ -0,0 +1,38 @@
+import { deepStrictEqual } from "assert";
+import { Token, tokenize } from "./utils";
+
+describe("Namespace", () => {
+  it("simple global namespace", async () => {
+    const tokens = await tokenize("namespace Foo");
+    deepStrictEqual(tokens, [
+      Token.keywords.namespace,
+      Token.meta(" ", "namespace-statement"),
+      Token.identifiers.type("Foo"),
+    ]);
```

Comment:
@nguerrera using the same system as this the csharp tm language. https://github.com/dotnet/csharp-tmLanguage/blob/main/test/
I feel this will produce some more meaningful tests than the bicep snapshot way which is very hard to read/understand.

However a lot of our token right now seems to be quite different around the `meta` tokens.. See for example the namespace tests https://github.com/dotnet/csharp-tmLanguage/blob/e852170471492fc403511a02a6a15eb43dbdaa2f/test/namespace.tests.ts#L21

Wondering if we should align our grammar more toward those patterns or this is good.

Reply:
actually, seeing the comment about missing punctuation in the grammar it make sense. I think I'll try to add it as it will make the test much more meaningful.

Reply:
Yes, I always meant to go back and classify the punctuation.

Reply:
I agree these tests are looking nicer than the snapshot approach. It will be easier to review and author. Good work.